packages/flutter_test/lib/src/matchers.dart

```diff
@@ -1090,3 +1098,57 @@ class _ClipsWithShapeBorder extends _MatchRenderObject<RenderClipPath, Null> {
   Description describe(Description description) =>
     description.add('clips with shape: $shape');
 }
+
+class _CoversSameAreaAs extends Matcher {
+  const _CoversSameAreaAs(
+    this.expectedPath,
+    {
+      @required this.areaToCompare,
+      this.numSamplesSqrt = 20,
+    }
+  );
+
+  final Path expectedPath;
+  final Rect areaToCompare;
+  final int numSamplesSqrt;
+
+  @override
+  bool matches(covariant Path actualPath, Map<dynamic, dynamic> matchState) {
+    for (int i = 0; i < numSamplesSqrt; i = i+1) {
+      for (int j = 0; j < numSamplesSqrt; j = j+1) {
+        final Offset offset = new Offset(
+          i * (areaToCompare.width / numSamplesSqrt),
+          j * (areaToCompare.height / numSamplesSqrt)
+        );
```

Comment:
it's important to also check outside the main area, e.g. in case the path has an extra box at the bounding box that flips the entirety of the "outside" to be "inside" by accident. Look at the unit circle matcher for inspiration of numbers to test.

Reply:
If the test author wants to sample a bigger box he can just pass a bigger box.
IMHO an API with "sample area" is more clear than an API with "focus area".
The test author should have the relevant knowledge to decide which sampling box to use.

Reply:
As discussed, we're leaving this for the test author. I added a note about this in the method's doc comment.